src/run_program.rs

```diff
@@ -99,11 +105,11 @@ struct RunProgramContext<'a, D> {
     softfork_stack: Vec<SoftforkGuard>,
     #[cfg(feature = "counters")]
     pub counters: Counters,
-
     #[cfg(feature = "pre-eval")]
     pre_eval: Option<PreEval>,
     #[cfg(feature = "pre-eval")]
     posteval_stack: Vec<Box<PostEval>>,
+    collected_ops: Option<Vec<CollectedOp>>,
```

Comment:
I do have reservations about making more complex this code which is already a bit too complex and nasty. You might be able to implement this functionality with `pre_eval`, and you can surely do it by creating a new separate "wallet/debugging" dialect where you can override the opcodes you need.

Reply:
Good point, a new dialect might be cleaner

Reply:
I've done a similar thing with the python api without mutable dialects [here](https://github.com/richardkiss/clvm_rs/commit/5945a5649e1591b8118646f9df9e52ce4f484f4d), but I guess `pyo3` is essentially `RefCell`-like objects everywhere. Might be worth trying it both ways: mutable dialects and with a `RefCell` to see what is easiest. `RefCell` is usually kind of a bad smell because if you make a mistake, you get a runtime error instead of a compile error. But I think in this case, it should be pretty easy to avoid a runtime error so `RefCell` might be the least intrusive without being *too* gross.